Converted EPS testing pseudocode

Vysares opened the pull request:
Turned Jashan's pseudocode into real code. Watchdog functionality included.  I kept these testing scripts in one file each so we can just throw it into an arduino sketch and upload it, and so it is easy to make edits during testing if we need to. For the actual FSW the watchdog feeding will be handled in essentially the same way, but in one of the util files.

@JashanChopra Please let me know if I messed up any pin numbers.

Vysares committed:
converted EPS testing pseudocode

Turned Jashan's pseudocode into real code. Watchdog functionality included.

Vysares committed:
Update watchdog.cpp

spelled watchdog correctly.

Vysares committed:
Event Sytem Initial commit

Used for inter-module communication and timing

JashanChopra commented on FSW/headers/eventUtil.hpp:
```diff
@@ -0,0 +1,52 @@
+
+/* - - - - - - Class Definitions - - - - - - */
+
+class Event
```
It would help if we write up either some external documentation or write some general code comments in here about what the classes do. I think classes can get confusing when there ends up being a lot to go track.

JashanChopra commented on Test/default.cpp:
```diff
@@ -8,47 +8,108 @@
+
+/* - - - - - - Initialization - - - - - - */
+// pin numbers
+const int PIN_AREG_CURR = 17;   // analog regulator current pin
```
Pins good

JashanChopra commented on Test/default.cpp:
```diff
@@ -8,47 +8,108 @@
+/* - - - - - printStatus - - - - - */
+void printStatus()
+{
+  float aCurr = HI_VOLTAGE*(float)analogRead(PIN_AREG_CURR)/(ADC_BINS*R_SENSE);   // analog regulator current, Amps
```
I think the bin conversion (3.3 / 1024) should be a constant for clarity.

JashanChopra replied:
Also, the current (both analog and digital) is actually the same as the voltage reading of the pin, because the voltage is in units of 1 V/A. I did the math to show Jackson, but I can also show you later Jaret. I'll manually determine any error in the lab.

JashanChopra commented on Test/default.cpp:
```diff
@@ -8,47 +8,108 @@
+  pinMode(PIN_AREG_CURR, INPUT);
+  pinMode(PIN_DREG_CURR, INPUT);
+  pinMode(PIN_DREG_PG, INPUT);
+  pinMode(PIN_WD_RESET,OUTPUT);
```
Space after comma fit formatting above.

JashanChopra commented on Test/lightCollect.cpp:
```diff
@@ -1,49 +1,105 @@
+/* - - - - - - Initialization - - - - - - */
+// pin numbers
+const int PIN_ADC_CS = 10;      // ADC chip select pin
+const int PIN_PHOTO = 14;       // direct photodiode data pin for use with Teensy ADC
```
PIN_PHOTO should be 21.

JashanChopra commented on Test/lightCollect.cpp:
```diff
@@ -1,49 +1,105 @@
+/* - - - - - readDataBackup - - - - - */
+void readDataBackup() // read data direct from pin using Teensy ADC
+{
+    double sciData = HI_VOLTAGE*(float)analogRead(PIN_PHOTO)/ADC_BINS;
```
Convert HI_VOLTAGE / ADC_BINS to a constant here as well.

JashanChopra reviewed (approved):
few small changes necessary, looks excellent

Vysares committed:
Updated EPS test scripts

changed watchdog interval to 100ms.
AnalogRead now uses a voltae resolution constant and current readings should be correct.
Added space before comma.
